### PR TITLE
Update collect env details and issue template

### PR DESCRIPTION
**Pull request by luca-medeiros**
## What does this PR do?
This PR expands the depth of detail collected by requirements/collect_env_details.py.
Before it collected selected libraries only. Now, with this PR all installed packages are displayed and all torch/lightning related are also filtered.

Due to the size of the output, added a toggle sample to the issue template.

Discussed with @carmocca here https://github.com/Lightning-AI/lightning/pull/13815#discussion_r929967403.
Related PR https://github.com/Lightning-AI/lightning/pull/13857

New output sample:
image

Fixes https://github.com/Lightning-AI/lightning/issues/13747

## Before submitting

- [X] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [X] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [X] Is this pull request ready for review? (if not, please submit in draft mode)
- [X] Check that all items from **Before submitting** are resolved
- [X] Make sure the title is self-explanatory and the description concisely explains the PR
- [X] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -41,7 +41,15 @@ You can get the script and run it with:
 ```bash
 wget https://raw.githubusercontent.com/Lightning-AI/lightning/master/requirements/collect_env_details.py
 python collect_env_details.py
+
 ```
+And paste it below inside the Details summary toggle.
+
+<details>
+  <summary>Details</summary>
+
```
Comment (awaelchli) [inline .github/ISSUE_TEMPLATE/bug_report.md:50]: Maybe we should put the indicator text exactly here where they should paste the content?

**Review by awaelchli — APPROVED** [review]
Nice

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (luca-medeiros) [reply .github/ISSUE_TEMPLATE/bug_report.md:50]: @awaelchli Agree!
But at the same time, keeping the toggle inside the comment block might be not very intuitive...
A user can paste the results inside the toggle and forget to move the toggle outside the comment block.

**Review by akihironitta — APPROVED** [review]
🚀

**Review by Borda — APPROVED** [review]

**Comment by awaelchli** [comment]
When I reviewed this PR, I misunderstood the goal here. I thought that we filtered only the ones that are related to lightning. However, it turns out that it dumps the list of the entire environment. This is not helpful, as in many cases all we need to know is the version of a few packages. The problem with the current way is that it creates an unreadable issue size, see the examples here:

https://github.com/Lightning-AI/lightning/issues/14218
https://github.com/Lightning-AI/lightning/issues/14198

I would be very thankful if we could reconsider the approach for practical reasons, thanks!

**Comment by luca-medeiros** [comment]
@awaelchli 
I had similar concerns when discussing with @carmocca here https://github.com/Lightning-AI/lightning/pull/13815#discussion_r930564557
But as he explained, having more info about the user environment can do be helpful in diagnosing bugs. 
Would putting that information on a separate toggle help with visibility issue?

I like the requirements files idea. We could append PyTorch Lightning and APP base requirements and print only those.

After the team decides what would be the best approach I can open a PR for it!

**Comment by awaelchli** [comment]
Where is the evidence that this additional information either helped us in the past or prevented us from helping when not provided @carmocca ? I have resolved many issues in the past and 99.9% of all these have been resolved by knowing the PL and PT or whatever dependency the issue was about. Also, in practice, just taking the list of hard pinned dependency and trying to install into a fresh environment is easier said than done. If the information can be hidden in a toggle, that would be great, but it doesn't happen. People just paste the 100's of lines of dependencies into the issue raw text.

**Comment by justusschock** [comment]
I agree with @awaelchli . 
I tried to install the entire dump and since it came from another OS, I immediately ran into issues. 
It would be nice if the program would output markdown code including the toggle already so that we have PL, PT and TM metrics always (together with Python, OS and processor) and the others hidden in the toggle. Otherwise I always need to manually go through the requirements to filter.

**Comment by carmocca** [comment]
I think having a full environment display can be very helpful, especially considering that lightning apps are not part of this repository and their issues might include code using any packages that are not in our requirements.

But I'm fine with outputting a nice markdown toggle so the output doesn't clutter the issue description

**Pull request merged**